internals/overlord/state/state.go

```diff
@@ -87,12 +87,17 @@ type State struct {
 	lastTaskId   int
 	lastChangeId int
 	lastLaneId   int
+	lastNoticeId int
 
 	backend  Backend
 	data     customData
 	changes  map[string]*Change
 	tasks    map[string]*Task
 	warnings map[string]*Warning
+	notices  map[string]*Notice
+
+	noticeWaiters  map[int]noticeWaiter
+	noticeWaiterId int
 
```

Comment:
Should we think about overflow of long running pebble instances that add/remove waiters over time ?

Reply:
Yeah, I thought about this, but 1) the int max is 2^63 == 9,223,372,036,854,775,808, which isn't going to happen in the lifetime of the universe, and 2) it doesn't even matter if it does overflow, as long as the first waiter isn't still alive. Given that waiters have a timeout, it would be extremely unlikely to overflow even if this were a uint8. But as an int (int64 on most systems) we'll definitely be fine.

Reply:
Thanks for explaining, sounds good. Pebble will also run on 32bit, but that is still fine.